docs/src/physics/navier_stokes_and_tracer_conservation.md

```diff
@@ -16,3 +28,91 @@ user.
       for an oceanographic introduction to the Boussinesq equations and Vallis (2017, Section 2.A)
       for an asymptotic derivation. See Kundu (2015, Section 4.9) for an engineering
       introduction.
+
+The fluid density $\rho$ in Oceananigans.jl is, in general, decomposed into three
+components:
+```math
+    \rho(\bm{x}, t) = \rho_0 + \rho_*(z) + \rho'(\bm{x}, t) \, ,
+```
+where $\rho_0$ is a constant 'reference' density, $\rho_*(z)$ is a background density
+profile which, when non-zero, is typically associated with the hydrostatic compression
+of seawater in the deep ocean, and $\rho'(\bm{x}, t)$ is the dynamic component of density
+corresponding to inhomogeneous distributions of a buoyant tracer such as temperature or salinity.
+The fluid *buoyancy*, associated with the buoyant acceleration of fluid, is
+defined in terms of $\rho'$ as
+```math
+    b = - \frac{g \rho'}{\rho_0} \, ,
+```
+where $g$ is gravitational acceleration.
+
+The Boussinesq approximation is valid when $\rho_* + \rho' \ll \rho_0$, which implies the
+fluid is _approximately_ incompressible, and thus does not support acoustic waves.
+In this case, the mass conservation equation reduces to the continuity equation
+```math
+    \bm{\nabla} \bm{\cdot} \bm{u} = \partial_x u + \partial_y v + \partial_z w = 0 \, .
+    \tag{eq:continuity}
+```
+
+## The momentum conservation equation
+
+The equations governing the conservation of momentum in a rotating fluid, including buoyancy
+via the Boussinesq approximation and including the averaged effects of surface gravity waves
+at the top of the domain via the Craik-Leibovich approximation are
+```math
+    \partial_t \bm{u} = - \left ( \bm{u} \bm{\cdot} \bm{\nabla} \right ) \bm{u}
+                        - \left ( \bm{U} \bm{\cdot} \bm{\nabla} \right ) \bm{u}
+                        - \left ( \bm{u} \bm{\cdot} \bm{\nabla} \right ) \bm{U}
+                        - \left ( \bm{f} - \bm{\nabla} \times \bm{u}^S \right ) \times \bm{u}
+                        - \bm{\nabla} \phi
+                        + b \bm{\hat z}
+                        - \bm{\nabla} \bm{\cdot} \bm{\tau}
+                        + \partial_t \bm{u}^S
+                        + \bm{F_u} \, ,
+    \tag{eq:momentum}
+```
```

Comment:
Might be pretty intimidating to start with this momentum equation haha but nice work on updating the docs in this PR!